ui-core/src/components/inputs/TolerancePicker.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useState, useRef, useMemo } from 'react';
+import Input, { InputProps } from './Input';
+import Modal from '../Modal';
+
+export type ToleranceValues = {
+  minusTolerance: number;
+  plusTolerance: number;
+};
+
+type ToleranceRange = { label: string; value: number };
+
+export type TolerancePickerProps = InputProps & {
+  onToleranceChange: (toleranceValues: ToleranceValues) => void;
+  toleranceRanges: ToleranceRange[];
+  toleranceValues?: ToleranceValues;
+};
+
+const TolerancePicker: React.FC<TolerancePickerProps> = ({
+  onToleranceChange,
+  toleranceRanges,
+  toleranceValues: { minusTolerance, plusTolerance } = {
+    minusTolerance: toleranceRanges[0].value,
+    plusTolerance: toleranceRanges[0].value,
+  },
+  ...inputProps
+}) => {
+  const [showPicker, setShowPicker] = useState(false);
+  const [inputValue, setInputValue] = useState(
+    `-${toleranceRanges[0].label}/+${toleranceRanges[0].label}`
+  );
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const { minusToleranceIndex, plusToleranceIndex } = useMemo(() => {
+    const minusToleranceIndex = toleranceRanges.findIndex(
+      (range) => range.value === minusTolerance
+    );
+    const plusToleranceIndex = toleranceRanges.findIndex((range) => range.value === plusTolerance);
+    return { minusToleranceIndex, plusToleranceIndex };
+  }, [minusTolerance, plusTolerance, toleranceRanges]);
+
+  const handleTimeChange = (
+    { label: newMinusLabel, value: newMinusValue }: ToleranceRange,
+    { label: newPlusLabel, value: newPlusValue }: ToleranceRange
+  ) => {
+    const formattedValue = `-${newMinusLabel}/+${newPlusLabel}`;
+    setInputValue(formattedValue);
```

Comment:
We should update the inputValue when the toleranceValues changes. Can you add a useEffect which update the inputValue each time the toleranceValues changes ?
```suggestion
```